python/ray/util/metrics.py

```diff
@@ -29,8 +36,9 @@ def __init__(
         description: str = "",
         tag_keys: Optional[Tuple[str, ...]] = None,
     ):
-        if len(name) == 0:
-            raise ValueError("Empty name is not allowed. Please provide a metric name.")
+        # Metrics with invalid names will be discarded and will not be collected
+        # by Prometheus
+        self.discard_metric = self._is_invalid_metric_name(name)
```

Comment:
```suggestion
        self._discard_metric = self._is_invalid_metric_name(name)
```
all private attributes should be underscore-prefixed by default (keeps the namespace of the python object clean)